Test cases for validating docker run options

sangeethah opened the pull request:
@ibuildthecloud 

Test cases for validating docker run option - priviledged (on/off), publishAllPorts and dataVolumes.

Test case creates a container with the options- priviledged (on/off), publishAllPorts and dataVolumes . Logs in to the host machine where the containers are running and validates that these options are set on the container by using the docker inspect command.

cloudnautique commented on tests/validation/requirements.txt:
```diff
@@ -4,4 +4,5 @@ pytest
 requests
 cattle
 selenium
+paramiko
```
looks like it is in here twice. Need to remove one.

cloudnautique commented on tests/validation/cattlevalidationtest/core/test_container_run_option.py:
```diff
@@ -0,0 +1,275 @@
+    c = client.wait_success(c)
+    assert c.state == "running"
+
+    ssh = get_ssh_to_host_ssh_container(host)
```
Lets use the python `with` statement to handle closing the ssh connection.

cloudnautique commented on tests/validation/cattlevalidationtest/core/test_container_run_option.py:
```diff
@@ -0,0 +1,275 @@
+
+    configs = check_docker_param_values(ssh, c, configs)
+
+    ssh.close()
```
Might need to use the contextlib closing method.

sangeethah commented:
@cloudnautique , I have submitted a revision which includes the following changes:

1. Removed paramiko from requirements.txt
2. Added the following code for using the  ssh container validation in test_container_run_option.py

with get_ssh_to_host_ssh_container(host) as ssh:
+        configs = check_docker_param_values(ssh, c, configs)
3. Made the host ssh container creation to be done in parallel on the hosts
4. Using the image - rancher/ssh-host-container:v0.1.0 for the host ssh container.

cloudnautique commented:
Overall, these LGTM. The tests run, and the SSH implementation runs. It looks like the second test case is failing because we are unable to publish all ports. If we can get to the bottom of that issue, we should be good to merge these. 

ping @ibuildthecloud

sangeethah commented:
Second test case - "test_container_run_with_options_2" tested for launching containers with publishAllPorts=True with managed network. In this case , we do expect that the publishAllPorts docker flag is not set. But instead provide the random port mapping for all the exposed port using iptables. This is broken and is tracked in  #196.
 
I have modified the test case  for the following use case : make sure that when publishAllPorts is set to True for unmanaged networks , the publishallport flag is set on the container using docker inspect command.

alena1108 commented on tests/validation/cattlevalidationtest/core/test_container_run_option.py:
```diff
@@ -0,0 +1,269 @@
+            print line
+            if docker_param_value in line.rstrip('\n'):
+                res = True
+            break
```
Should move "break" under "if"

alena1108 commented on tests/validation/cattlevalidationtest/core/test_container_run_option.py:
```diff
@@ -0,0 +1,269 @@
+                        % (str(config["docker_param_name"]),
+                           str(config["docker_param_value"]),
+                           str(config["docker_value"]))
+        print result_string
```
We probably don't need "print" here as we only care when something goes wrong

sangeethah commented:
@alena1108 , I have done the following fixes as per your review comments:

1. Removed all the print statements
2. Moved break statement to be part of if statement.

alena1108 commented:
LGTM

cloudnautique commented:
:shipit:
